### PR TITLE
Component intialization

### PR DESCRIPTION
**Pull request by NANAYEMYINTKYI**
Most of the component is done but filter interpreters still need to be discussed and connect with backend.

### CODE REVIEW

File: client/src/components/BtnDownload.jsx
Comment (Chatchaponz) [inline client/src/components/BtnDownload.jsx:1]: File
- BtnDownload
- BtnFind
- BtnOtherword
Can it group into single components?
Somethin like this
```JavaScript
function CustomBtn({label}) {
  return (
    <button className="rounded bg-secondary-base py-2 px-4 text-sm text-white hover:bg-primary-base active:bg-sky-700">
      {label}
    </button>
  );
}

export default BtnDownload;
````

And do something like this when use
```javascript
<CustomBtn  label="ดาวน์โหลด"/>
...
```

But if there are other reasons, why these components need to be separated, pls provide those reason as well na.

File: client/src/components/FilterCategory.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import { useState } from 'react'
+import { Combobox } from '@headlessui/react'
+
+const categories = [
+    //fetch from backend
```
Comment (Chatchaponz) [inline client/src/components/FilterCategory.jsx:5]: Waiting...

File: client/src/components/Searchbar.jsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import { CheckIcon } from '@heroicons/react/20/solid';
+
+function SearchBar() {
+  const [searchResults, setSearchResults] = useState([]); // hold search results
+  const [selectedSearch, setSelectedSearch] = useState(null); // hold the selected item
+  const [query, setQuery] = useState(''); // hold the search query
+
+  // Fetch data from backend
+  useEffect(() => {
+    const fetchSearchResults = async () => {
```
Comment (Chatchaponz) [inline client/src/components/Searchbar.jsx:12]: Pls move this out of `useEffect()`

File: client/src/components/Searchbar.jsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      try {
+        const response = await fetch('/api/search'); // Replace with your API endpoint
+        const data = await response.json();
+        setSearchResults(data);
+      } catch (error) {
+        console.error('Error fetching search results:', error);
+      }
+    };
```
Comment (Chatchaponz) [inline client/src/components/Searchbar.jsx:20]: Pls move this out of `useEffect()`

File: client/src/pages/home/HomePage.jsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export const Home = () => {
+  const [message, setMessage] = useState("");
+
+  //  useEffect(() => {
+  //    fetch("/api")
+  //      .then((response) => response.json())
+  //      .then((data) => setMessage(data.message));
+  //  }, []);
+  //  return <div className="mt-10 flex w-full flex-col">{message}</div>;
```
Comment (Chatchaponz) [inline client/src/pages/home/HomePage.jsx:13]: Remove unused comment na

**Review by Chatchaponz — CHANGES_REQUESTED** [review]
Overall, every thing look OK with minor adjustment

File: client/src/pages/admin/components/FilterStatus.jsx
Comment (wutpppppppppppppppppp) [inline client/src/pages/admin/components/FilterStatus.jsx:1]: Different design as it serves as a filter component? please give a feedback or a reason supporting the differences of the design

File: client/src/pages/home/HomePage.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import Navbaruser from "../../components/Navbaruser";
+
+export const Home = () => {
+  const [message, setMessage] = useState("");
+
+  //  useEffect(() => {
+  //    fetch("/api")
+  //      .then((response) => response.json())
```
Comment (wutpppppppppppppppppp) [inline client/src/pages/home/HomePage.jsx:10]: Remove unused code

**Review by wutpppppppppppppppppp — CHANGES_REQUESTED** [review]
# Overall
slight adjustment for furthur clarification and easy maintenance.

File: client/src/components/FilterInterpreter.jsx [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import {useState} from React
+import { Combobox, ComboboxInput, ComboboxOption, ComboboxOptions } from '@headlessui/react'
```
[the remark this answers is not in the document]
Comment (NANAYEMYINTKYI) [reply client/src/components/FilterInterpreter.jsx:3]: Roger , fix it to be in the same pattern kub

**Review by wutpppppppppppppppppp — COMMENTED** [review]
Okay na krub

**Review by wutpppppppppppppppppp — APPROVED** [review]

File: client/src/components/FilterInterpreter.jsx [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+const interpreters= [
+  {id: 1 ,name: 'ล่ามโทรทัศน์'},
+  {id: 2 ,name: 'ล่ามชุมชน'},
+  {id: 3 ,name: 'คนหูดี'},
+  {id: 4 ,name: 'คนหูตึง'},
+  {id: 5 ,name: 'คนหูหนวก'},
+]
+
+function FilterInterpreter(){
+  const [SelectInterpreter, setSelectInterpreter] = useState(interpreters[0])
```
Comment (Chatchaponz) [inline client/src/components/FilterInterpreter.jsx:14]: Should be `selectInterpreter`???

File: client/src/pages/home/HomePage.jsx [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import Navbaruser from "../../components/Navbaruser";
+
+export const Home = () => {
+  const [message, setMessage] = useState("");
+
+  return (
+   <Navbaruser></Navbaruser>
```
Comment (Chatchaponz) [inline client/src/pages/home/HomePage.jsx:9]: `<Navbaruser/>` might be better if component have no children

**Review by Chatchaponz — COMMENTED** [review]
About some minor coding standard

File: client/src/components/FilterWordtype.jsx
[the remark this answers is not in the document]
Comment (NANAYEMYINTKYI) [reply client/src/components/FilterWordtype.jsx:1]: Fix the pattern of the wordtype to be listbox and change all to comboinput and combooption kub.

File: client/src/components/FilterInterpreter.jsx (hunk 7 above)
Comment (NANAYEMYINTKYI) [reply client/src/components/FilterInterpreter.jsx:14]: yes kub , fix it.

File: client/src/pages/home/HomePage.jsx (hunk 8 above)
Comment (NANAYEMYINTKYI) [reply client/src/pages/home/HomePage.jsx:9]: fix it ka and I move the component to the router file , so it don't need to be import every time we create new page.

**Review by Chatchaponz — APPROVED** [review]

**Pull request merged**